acme/acme/challenges.py

```diff
@@ -216,6 +219,81 @@ def response_and_validation(self, account_key, *args, **kwargs):
 
 
 @ChallengeResponse.register
+class DNS01Response(KeyAuthorizationChallengeResponse):
+    """ACME "dns-01" challenge response."""
+    typ = "dns-01"
+
+    def simple_verify(self, chall, domain, account_public_key):
+        """Simple verify.
+
+        :param challenges.DNS01 chall: Corresponding challenge.
+        :param unicode domain: Domain name being verified.
+        :param account_public_key: Public key for the key pair
+            being authorized. If ``None`` key verification is not
+            performed!
+        :param JWK account_public_key:
+
+        :returns: ``True`` iff validation is successful, ``False``
+            otherwise.
+        :rtype: bool
+
+        """
+        if not self.verify(chall, account_public_key):
+            logger.debug("Verification of key authorization in response failed")
+            return False
+
+        validation_domain_name = chall.validation_domain_name(domain)
+        validation = chall.validation(account_public_key)
+        logger.debug("Verifying %s at %s...", chall.typ, validation_domain_name)
+        try:
+            dns_response = dns.resolver.query(validation_domain_name, 'TXT')
+            txt_records = sum([rdata.strings for rdata in dns_response], [])
+        except dns.exception.DNSException as error:
+            logger.error("Unable to resolve %s: %s", validation_domain_name,
+                         error)
+            return False
+
+        for txt_record in txt_records:
+            if txt_record == validation:
+                return True
```

Comment:
Agreed about containing `dnspython`. However, too many loops for my taste. This would be more Pythonic and likely more efficient:

```python
exists = validation in txt_records
if not exists:
  logger.debug(...)
return exists
```